src/telemetry/metric_registry.cc

```diff
@@ -107,6 +109,50 @@ metric_registry::~metric_registry() {
 
 namespace {
 
+// -- free functions used by impl_base::collect()
+
+label_list extract_labels(const ct::metric* instance,
+                          std::vector<label_view>& vec) {
+  auto get_value = [](const auto& label) -> label_view {
+    auto name = label.name();
+    auto val = label.value();
+    return {std::string_view{name.data(), name.size()},
+            std::string_view{val.data(), val.size()}};
+  };
+  vec.clear();
+  const auto& labels = instance->labels();
+  std::transform(labels.begin(), labels.end(), std::back_inserter(vec),
+                 get_value);
+
+  return vec;
```

Comment:
Having an output parameter *and* returning it feels quite redundant (and can obfuscate the call side). I think this should simply return `void`.

```suggestion
```